---
title: "BiocPyInterop introduction"
author: Vince Carey^[stvjc@channing.harvard.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{BiocPyInterop introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Workshop Title

Authors:
    Vince Carey^[Harvard Medical School],
    <br/>


## Overview

### Description

Abstract: Multilingual data science strategies can increase efficiency of discovery by taking advantage of diverse data management and analysis strategies.  

In this workshop we will examine interplay between R, Python, and Apache Spark in genetic and single-cell applications.  CITE-seq studies simultaneously quantify surface protein and mRNA abundance in single cells.  We will use scviR to compare interpretations based on deep learning and sequential component-specific methods.  

The UK Biobank is the foundation of thousands of genome-wide association studies.  The Telomere-to-Telomere project produced the first gapless human reference genome.  Both of these resources will be explored using BiocHail.  Workshop attendees will acquire an understanding of Aaron Lun's basilisk package and its use in isolating specific collections of python modules, the anndata representations and scvi-tools analyses of CITE-seq data, and the hail.is approach to structuring and analyzing massive genetics data resources using Spark Resilient Distributed Data.  All programming will be carried out in R; quarto documents that mix R and python will also be illustrated.

### Pre-requisites

* Basic knowledge of R syntax
* Interest in single-cell genomics, human genetics, deep learning

It will be helpful to have an acquaintance with 

* [a chapter of the OSCA book](http://bioconductor.org/books/3.17/OSCA.advanced/integrating-with-protein-abundance.html)
* [an scviR vignette](https://bioconductor.org/packages/release/bioc/vignettes/scviR/inst/doc/citeseq_tut.html)
* [a BiocHail vignette](https://bioconductor.org/packages/release/bioc/vignettes/BiocHail/inst/doc/gwas_tut.html)

### Participation

This is a 90 minute workshop that will cover

- programming with basilisk to establish predictable python infrastructure and interoperation
- exploration of torch-based tooling for single-cell analysis of a CITE-seq experiment
- exploration of spark-based tooling for interaction with 1000 genomes genotypes (and, if time permits, UK Biobank phenotypes)

### _R_ / _Bioconductor_ packages used

- basilisk
- OSCA.advanced
- scviR
- BiocHail

### Time outline

TBD

<!--
An example for a 45-minute workshop:

| Activity                     | Time |
|------------------------------|------|
| Packages                     | 15m  |
| Package Development          | 15m  |
| Contributing to Bioconductor | 5m   |
| Best Practices               | 10m  |

### Workshop goals and objectives

List "big picture" student-centered workshop goals and learning
objectives. Learning goals and objectives are related, but not the
same thing. These goals and objectives will help some people to decide
whether to attend the conference for training purposes, so please make
these as precise and accurate as possible.

*Learning goals* are high-level descriptions of what
participants will learn and be able to do after the workshop is
over. *Learning objectives*, on the other hand, describe in very
specific and measurable terms specific skills or knowledge
attained. The [Bloom's Taxonomy](#bloom) may be a useful framework
for defining and describing your goals and objectives, although there
are others.

### Learning goals

Some examples:

* describe how to...
* identify methods for...
* understand the difference between...

### Learning objectives

* analyze xyz data to produce...
* create xyz plots
* evaluate xyz data for artifacts

## Workshop

Divide the workshop into sections (`## A Section`). Include
fully-evaluated _R_ code chunks. Develop exercises and solutions, and
anticipate that your audience will walk through the code with you, or
work on the code idependently -- do not be too ambitious in the
material that you present.

-->



